bodhi/server/util.py

```diff
@@ -684,3 +684,35 @@ def sort_severity(value):
     }
 
     return value_map.get(value, 99)
+
+
+def infer_content_class(base, build):
+    """ Given a base class and a build from koji, identify and return the child
+    class associated with the appropriate ContentType.
+
+    For example, given the Package base class and a normal koji build, return
+    the RpmPackage model class. Or, given the Build base class and a container
+    build, return the ContainerBuild model class.
+
+    Arguments:
+    base -- A base model class.
+    build -- A dict of information from the build system (koji).
+    """
+
+    # Import this here to avoid circular import
+    from bodhi.server.models import ContentType
+
+    if 'extra' not in build:
+        raise KeyError("No extra block found for build %r" % build)
```

Comment:
It's my understanding that *old* versions of koji may not return an `extra` field... but the current versions do always return it.

I'll just remove the conditional, since testing it would require gymnastic mocking, and without it we'll still raise a KeyError anyways.